### DISCUSSION TITLE
Since things like buttons can end up with an unknown state, why track that?

### DISCUSSION THREAD

**Discussion by flyize** [jazzyisj/unavailable-entities-sensor#31] [Q&A] [answered]
I'm thinking of just removing unknown from the script, but was hoping for a sanity check in case I missed something. I've got like 40 buttons that currently have an unknown state...

**Comment by jazzyisj** [comment]
The template in the package is purposely kept as simple as possible.  Buttons only have an unknown state if they have never been pressed.  (There are some integrations that do seem to "lose" the last pressed state, but that has nothing to do with this sensor other than catching and highlighting that "bug").

It is absolutely valid and possible to ignore the button domain if you wish.  This very example is given in the readme.

https://github.com/jazzyisj/unavailable-entities-sensor#customizing-the-sensor

**Comment by flyize** [comment]
@jazzyisj Since I assume you know more than me about this, I'm just asking what you think I could miss by excluding 'unknown'. AFAIK, those are devices that HA is communicating with, but doesn't know their state. I'm only concerned about entities that its not communicating with, so in my case I'd think I can just get rid of 'unknown'.

Is my thought process correct? Or is there a reason you're tracking 'unknown' that I don't yet understand?

**Comment by jazzyisj** [comment, marked as answer]
One example.  A template select entity is not able to evaluate to one of the available selections for some reason.  The entity resolves to the state of "unknown".  You might want to know about this.  I have this situation right now because Music Assistant is crapped out until it is updated.  So all of my music playlist preset selections are "unknown".  I want to know about this because none of my presets will play (including alarm clocks).

There are many other examples.  If there is nothing in your own config that might have a state of unknown that you care about then by all means alter the template to suit your needs!

The template in the package is more or less just a starting point.  Personally I have several different versions of it in my own config that tell me different things.

Here's an example of one that just tells me if any [media players are unavailable](https://github.com/jazzyisj/home-assistant-config/blob/377cc5d10a644f291d85ea60ee132b89b03576ed/templates/media/template_media_players.yaml#L57).
